pkg/server/server.go

```diff
@@ -92,6 +92,12 @@ var (
 		NativeHistogramMaxBucketNumber:  100,
 		NativeHistogramMinResetDuration: time.Hour,
 	}, []string{"grpc_service", "grpc_method", "datastore_query_count", "dispatch_count", "consistency"})
+
+	throttledRequestCounter = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
Do we want to break it down by grpc_method (so that we can distinguish between throttling in list user vs. other API in the future)?

Reply:
I thought the same but it's a counter, can you add labels on a counter? 🤔

Reply:
It's a valid question. As an operator, you're going to have different throttling configurations for the different APIs so separating into different counters may be useful. On the other hand, we're going to need to expand to also include counts for timed-out requests which would double the number of counters. Having six counters, two for each API, may become bloated for a niche metric.

Reply:
Maybe a counter isn the right type of metric?

Reply:
I thought counter metrics can have [attributes](https://opentelemetry.io/docs/specs/otel/metrics/api/#add)

Reply:
Counter is the right type: https://prometheus.io/docs/concepts/metric_types/

I'm in favor of distinguishing by service and method, like the other metrics.

Reply:
See https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec

Reply:
can we call it `grpc_service` so that it has similar feel as other stats that are differentiated by API?

Reply:
likely we also want to have grpc_service" as well to have the same look and feel as other API

Reply:
@adriantam you mean `grpc_method`, right?